session/session.go

```diff
@@ -107,15 +117,20 @@ func retrieveSession(ctx context.Context, cfg *config.Config,
 	r, err := c.User.GetSelf(p, nil)
 	if err != nil {
 		switch err.(type) {
+		case *user.GetSelfBadRequest:
+			return &Authorized{fromEnvVars: fromEnvVars, isTeam: true}, nil
+
 		case *user.GetSelfUnauthorized:
 			// Stored token is not valid
 
 			// Clear stored token
-			cfg.AuthToken = ""
-			err = cfg.Write()
-			if err != nil {
-				return nil, hierr.Errorf(err,
-					"Failed to update config after clearing expired auth token.")
+			if cfg.AuthToken != "" {
```

Comment:
Why add this logic?

Reply:
I removed a case above that was skipping writing to disk if the token was empty, and don't want to always write to disk if there's no need.

Reply:
gotcha :+1: